src/pages/create/Activity.tsx

```diff
@@ -0,0 +1,282 @@
+        client,
+        "SELECT Id, Name FROM User WHERE UserPermissionsSFContentUser=true",
+        200
+      ),
+    {
+      enabled: !!activityTypes
+        .find((e) => e.value === type)
+        ?.fields.findIndex((e) => e.name === "OwnerId"),
+    }
+  );
+
+  const ActivityMetadata = useQueryWithClient(
+    [QueryKey.ACTIVITY_METADATA, type],
+    (client) =>
+      getObjectMeta(
+        client,
+        activityTypes.find((e) => e.label === type)?.value as string
+      ),
+    {
+      enabled: !!type,
+    }
+  );
+
+  useEffect(() => {
+    if (!type || !ActivityMetadata.data) return;
+
+    const activityType = activityTypes.find((e) => e.label === type);
+
+    setSchema(
+      getActivitySchema(
+        activityType?.fields as Field[],
+        activityType?.value as string
+      )
+    );
+    setValue(`${activityType?.value}Subtype`, activityType?.label);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [ActivityMetadata.data, watch, type]);
+
+  const submit = async (data: any) => {
+    if (!client) return;
+
+    setIsSubmitting(true);
+
+    delete data.Type;
+
+    if (data.EventSubtype) {
+      data.DurationInMinutes = Math.abs(
+        (new Date(data.EndDateTime as string).getTime() -
+          new Date(data.ActivityDateTime as string).getTime()) /
+          1000 /
+          60
+      );
+    }
+
+    await postData(
+      client,
+      activityTypes.find((e) => e.label === type)?.value as string,
+      data
+    );
+
+    navigate(-1);
+  };
+
+  return (
+    <form onSubmit={handleSubmit(submit)} style={{ margin: "5px" }}>
+      <DropdownSelect
+        title="Type"
+        value={type || ""}
+        onChange={(e) => {
+          setValue("Type", e);
+        }}
+        error={!!errors.Type}
+        data={activityTypes}
+        keyName={"label"}
+        valueName={"label"}
+      />
+      {type && (
+        <Stack vertical gap={12}>
+          {activityTypes
+            .find((e) => e.label === type)
+            ?.fields.map((field, i) => {
+              let values: { value: string; key: string }[] = [];
+
+              if (field.label === "Type") return;
+
+              if (field.type === "dropdown") {
+                switch (field.name) {
+                  case "Status":
+                    values = ActivityMetadata.data?.fields
+                      ?.find((e) => e.label === "Status")
+                      ?.picklistValues.filter((e) => e.active)
+                      .map((e) => ({ key: e.label, value: e.value })) as {
+                      value: string;
+                      key: string;
+                    }[];
+                    break;
+                  case "OwnerId":
+                    values = people.data?.map((e) => ({
+                      key: e.Id,
+                      value: e.Name,
+                    })) as { value: string; key: string }[];
+                }
+              }
+              switch (field.type) {
```

Comment:
I think salesforce field type -> field component should be a generic thing (for when we implement CRUD on other things). Could probably extract this into it's own component used by SObject CRUD forms